src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -224,6 +230,39 @@ void MainWindow::open_and_render_project(const QString& filepath, const QString&
     open_project(filepath);
 }
 
+void MainWindow::save_project(QString filepath)
+{
+    if (!filepath.isEmpty())
+    {
+        const QString Extension = "appleseed";
+
+        if (QFileInfo(filepath).suffix() != Extension)
+            filepath += "." + Extension;
+
+        filepath = QDir::toNativeSeparators(filepath);
+
+        if (m_project_file_watcher)
+            stop_monitoring_project_file();
+
+        m_project_manager.save_project_as(filepath.toAscii().constData());
+
+        if (m_project_file_watcher)
+            start_monitoring_project_file();
+
+        update_workspace();
+    }
+}
+
+Project* MainWindow::current_project()
+{
+    return m_project_manager.get_project();
+}
+
+bool MainWindow::is_project_dirty()
```

Comment:
maybe change all appearances of m_project_manager.is_project_dirty in MainWindow on this then?